File: backend/src/verification.ts
import { jwtVerify } from "utils/jtw"

export const verify = (token: string): boolean => {
  try {
    const jwt = jwtVerify(token);
    if (typeof jwt === "string") return false;
    // expiration is checked automatically
    // can check jwt['user_metadata']

    return true;
  } catch {
    return false;
  }
}

File: backend/src/verification.py
from utils.jwt import jwtVerify

def verify(token: str) -> bool:
    try:
        jwt = jwtVerify(token)
        if isinstance(jwt, str):
            return False
        # Expiration is checked automatically
        # Can check jwt['user_metadata']

        return True
    except:
        return False